src/riak_cs_pbc.erl

```diff
@@ -22,24 +22,65 @@
 
 -module(riak_cs_pbc).
 
--export([get_object/4,
-         put_object/5,
-         put/3,
+-export([ping/3,
+         get/6,
+         repl_get/7,
          put/4,
-         put_with_no_meta/2,
-         put_with_no_meta/3,
-         list_keys/3,
-         get_cluster_id/1,
+         put/5,
+         delete_obj/5,
+         mapred/5,
+         list_keys/4,
+         get_index_eq/5,
+         get_index_eq/6,
+         get_index_range/7,
+         get_cluster_id/1
+        ]).
+
+-export([
+         get_no_stats/5,
+         put_object/5,
+         put_no_stats/3,
+         put_no_stats/4,
+         list_keys_no_stats/3,
          check_connection_status/2]).
 
+-include_lib("riakc/include/riakc.hrl").
+-include("riak_cs_stats.hrl").
+
+-spec ping(pid(), timeout(), riak_cs_stats:key()) -> pong.
+ping(PbcPid, Timeout, StatsKey) ->
+    _ = riak_cs_stats:inflow(StatsKey),
+    StartTime = os:timestamp(),
+    Result = riakc_pb_socket:ping(PbcPid, Timeout),
+    case Result of
+        pong ->
+            _ = riak_cs_stats:update_with_start(StatsKey, StartTime);
+        _ ->
+            _ = riak_cs_stats:update_error_with_start(StatsKey, StartTime)
+    end,
+    Result.
+
 %% @doc Get an object from Riak
--spec get_object(pid(), binary(), binary(), proplists:proplist()|timeout()) ->
+-spec get_no_stats(pid(), binary(), binary(), proplists:proplist(), timeout()) ->
```

Comment:
After the discussion in chat, `s/_no_stats/_sans_stats/`.

 Note for self: don't forget replace repair script under priv/.